// Copyright 2019 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package tfx.components.bulkinferrer;

message ModelSpec {
  // Specifies the signature name to run the inference with. If multiple
  // signature names are specified (ordering doesn't matter), inference is done
  // as a multi head model. If nothing is specified, default serving signature
  // is used as a single head model.
  repeated string model_signature_name = 2;

  // Tags to select metagraph from the saved model. If unspecified, the default
  // tag selects metagraph to run inference on CPUs. See some valid values in
  // tensorflow.saved_model.tag_constants.
  repeated string tag = 5;

  reserved 1, 3, 4, 6;
}

message DataSpec {
  // Indicates which splits of examples should be processed.  If empty, then
  // all example splits are processed.  Example splits are IID subsets of
  // examples defined at the pipeline level with user defined names such as
  // "holdout", "training", "eval_only", "split0", and so on.
  repeated string example_splits = 2;

  reserved 1, 3, 4;
}
